Monitor name by accident includes health check command

#### Setup Details
 CIS Version : 2.9      
 Build: f5networks/k8s-bigip-ctlr:2.9.1       
 BIGIP Version: Big IP      
 AS3 Version: 3.38.0
 Agent Mode: AS3
 Orchestration: K8S
 Orchestration Version: 1.24.3
 Pool Mode: Nodeport      
 Additional Setup details: Platform UDF/CNI Cilium

#### Description  

Monitor name includes health check command (with space) that violates the object name pattern (without space).

#### Steps To Reproduce
Create a VS CRD in nginx-ingress namespace to trigger AS3 declaration sending from CIS to BIG-IP.
```
ubuntu@k8s-master:~$ cat echo-api-vs.yaml
apiVersion: "cis.f5.com/v1"
kind: VirtualServer
metadata:
  name: vs-echo-api
  namespace: nginx-ingress
  labels:
    f5cr: "true"
spec:
  host: echo.example.com
  hostGroup: "echo"
  ipamLabel: Test
  virtualServerHTTPPort: 80
  snat: auto
  pools:
  - monitor:
      interval: 20
      recv: ""
      send: "GET /api HTTP/1.1\r\nHost: echo.example.com\r\n"
      timeout: 10
      type: http
    path: /api
    service: nginx-ingress
    servicePort: 80
ubuntu@k8s-master:~$

```
#### Expected Result 

AS3 declaration is applied in BIG-IP.

#### Actual Result  

No configuration change in BIG-IP.

#### Diagnostic Information
CIS Debug Error Message:
```
2022/07/22 17:51:21 [ERROR] [AS3] Raw response from Big-IP: map[code:422 declarationFullId: errors:[/k8s/Shared: propertyName "nginx_ingress_nginx_ingress_http_80GET _api HTTP_1_1\r\nHost_ echo_example_com\r\n" should match pattern "^[A-Za-z]([0-9A-Za-z_.-]{0,188}[0-9A-Za-z_.])?$"] message:declaration is invalid] {"$schema":"https://raw.githubusercontent.com/F5Networks/f5-appsvcs-extension/master/schema/3.18.0/as3-schema-3.18.0-4.json","class":"AS3","declaration":{"class":"ADC","controls":{"class":"Controls","userAgent":"CIS/v2.9.1 K8S/v1.24.3"},"id":"urn:uuid:B97DFADF-9F0D-4F6C-8D66-E9B52E593694","k8s":{"Shared":{"class":"Application","crd_10_1_10_200_80":{"source":"0.0.0.0/0","translateServerAddress":true,"translateServerPort":true,"class":"Service_HTTP","virtualAddresses":["10.1.10.200"],"virtualPort":80,"snat":"auto","policyEndpoint":"/k8s/Shared/crd_10_1_10_200_80_echo_policy"},"crd_10_1_10_200_80_echo_policy":{"class":"Endpoint_Policy","rules":[{"name":"vs_echo_api_nginx_ingress_80_nginx_ingress_echo_example_com","conditions":[{"type":"httpHeader","name":"host","event":"request","all":{"values":["echo.example.com"],"operand":"equals"}},{"type":"httpUri","name":"1","event":"request","index":1,"pathSegment":{"values":["api"],"operand":"equals"}}],"actions":[{"type":"forward","event":"request","select":{"pool":{"use":"nginx_ingress_80_nginx_ingress_echo_example_com"}}}]}],"strategy":"first-match"},"nginx_ingress_80_nginx_ingress_echo_example_com":{"class":"Pool","members":[{"addressDiscovery":"static","serverAddresses":["10.1.1.5"],"servicePort":30948},{"addressDiscovery":"static","serverAddresses":["10.1.1.6"],"servicePort":30948}],"monitors":[{"use":"/k8s/Shared/nginx_ingress_nginx_ingress_http_80GET _api HTTP_1_1\r\nHost_ echo_example_com\r\n"}]},"nginx_ingress_nginx_ingress_http_80GET _api HTTP_1_1\r\nHost_ echo_example_com\r\n":{"class":"Monitor","interval":20,"monitorType":"http","targetAddress":"","timeout":10,"timeUntilUp":0,"adaptive":false,"dscp":0,"receive":"none","send":"GET /api HTTP/1.1\r\nHost: echo.example.com\r\n","targetPort":0},"template":"shared"},"class":"Tenant","defaultRouteDomain":0},"label":"CIS Declaration","remark":"Auto-generated by CIS","schemaVersion":"3.18.0"}}
2022/07/22 17:51:21 [ERROR] [AS3] Big-IP Responded with code: 422
```
#### Observations (if any)
N/A

Exact debug message:
```
2022/07/22 17:51:21 [ERROR] [AS3] Raw response from Big-IP: map[code:422 declarationFullId: errors:[/k8s/Shared: propertyName "nginx_ingress_nginx_ingress_http_80GET _api HTTP_1_1\r\nHost_ echo_example_com\r\n" should match pattern "^[A-Za-z]([0-9A-Za-z_.-]{0,188}[0-9A-Za-z_.])?$"] message:declaration is invalid] {"$schema":"https://raw.githubusercontent.com/F5Networks/f5-appsvcs-extension/master/schema/3.18.0/as3-schema-3.18.0-4.json","class":"AS3","declaration":{"class":"ADC","controls":{"class":"Controls","userAgent":"CIS/v2.9.1 K8S/v1.24.3"},"id":"urn:uuid:B97DFADF-9F0D-4F6C-8D66-E9B52E593694","k8s":{"Shared":{"class":"Application","crd_10_1_10_200_80":{"source":"0.0.0.0/0","translateServerAddress":true,"translateServerPort":true,"class":"Service_HTTP","virtualAddresses":["10.1.10.200"],"virtualPort":80,"snat":"auto","policyEndpoint":"/k8s/Shared/crd_10_1_10_200_80_echo_policy"},"crd_10_1_10_200_80_echo_policy":{"class":"Endpoint_Policy","rules":[{"name":"vs_echo_api_nginx_ingress_80_nginx_ingress_echo_example_com","conditions":[{"type":"httpHeader","name":"host","event":"request","all":{"values":["echo.example.com"],"operand":"equals"}},{"type":"httpUri","name":"1","event":"request","index":1,"pathSegment":{"values":["api"],"operand":"equals"}}],"actions":[{"type":"forward","event":"request","select":{"pool":{"use":"nginx_ingress_80_nginx_ingress_echo_example_com"}}}]}],"strategy":"first-match"},"nginx_ingress_80_nginx_ingress_echo_example_com":{"class":"Pool","members":[{"addressDiscovery":"static","serverAddresses":["10.1.1.5"],"servicePort":30948},{"addressDiscovery":"static","serverAddresses":["10.1.1.6"],"servicePort":30948}],"monitors":[{"use":"/k8s/Shared/nginx_ingress_nginx_ingress_http_80GET _api HTTP_1_1\r\nHost_ echo_example_com\r\n"}]},"nginx_ingress_nginx_ingress_http_80GET _api HTTP_1_1\r\nHost_ echo_example_com\r\n":{"class":"Monitor","interval":20,"monitorType":"http","targetAddress":"","timeout":10,"timeUntilUp":0,"adaptive":false,"dscp":0,"receive":"none","send":"GET /api HTTP/1.1\r\nHost: echo.example.com\r\n","targetPort":0},"template":"shared"},"class":"Tenant","defaultRouteDomain":0},"label":"CIS Declaration","remark":"Auto-generated by CIS","schemaVersion":"3.18.0"}}
2022/07/22 17:51:21 [ERROR] [AS3] Big-IP Responded with code: 422
```

--- a/docs/RELEASE-NOTES.rst
+++ b/docs/RELEASE-NOTES.rst
@@ -20,6 +20,7 @@ Added Functionality
         * :issues:`2361` Allow monitoring of an alias port in VirtualServer and TransportServer
         * :issues:`1933` Added serviceNamespace field in Pools for VirtualServer CR that allows to define a pool service from another namespace in a Virtual server CR.
           See `Examples <https://github.com/F5Networks/k8s-bigip-ctlr/tree/master/docs/config_examples/customResource/>`_
+        * :issues:`2507` Monitor name by accident includes health check command
         * Added TCP Health Monitor support for VS CRs
         * Added monitors support for VS and TS CRs
     * Ingress:

--- a/pkg/controller/resourceConfig.go
+++ b/pkg/controller/resourceConfig.go
@@ -288,19 +288,19 @@ func formatPoolName(namespace, svc string, port intstr.IntOrString, nodeMemberLa
 }
 
 // format the monitor name for an VirtualServer pool
-func formatMonitorName(namespace, svc string, monitorType string, port int32, sendString string) string {
+func formatMonitorName(namespace, svc string, monitorType string, port int32, hostName string, path string) string {
 	monitorName := fmt.Sprintf("%s_%s", svc, namespace)
+
+	if len(hostName) > 0 {
+		monitorName = monitorName + fmt.Sprintf("_%s", hostName)
+	}
+	if len(path) > 0 && path != "/" {
+		monitorName = monitorName + fmt.Sprintf("_%s", path)
+	}
 
 	if monitorType != "" && port != 0 {
 		servicePort := fmt.Sprint(port)
 		monitorName = monitorName + fmt.Sprintf("_%s_%s", monitorType, servicePort)
-	}
-	if len(sendString) > 0 && sendString != "/" {
-		if strings.Contains(sendString, "/") {
-			monitorName = monitorName + fmt.Sprintf("%s", sendString)
-		} else {
-			monitorName = monitorName + fmt.Sprintf("_%s", sendString)
-		}
 	}
 	return AS3NameFormatter(monitorName)
 }
@@ -437,7 +437,7 @@ func (ctlr *Controller) prepareRSConfigFromVirtualServer(
 			pool.MonitorNames = append(pool.MonitorNames, MonitorName{Name: pl.Monitor.Name, Reference: pl.Monitor.Reference})
 		} else if pl.Monitor.Send != "" && pl.Monitor.Type != "" {
 			if pl.Name == "" {
-				monitorName = formatMonitorName(vs.ObjectMeta.Namespace, pl.Service, pl.Monitor.Type, pl.ServicePort, pl.Monitor.Send)
+				monitorName = formatMonitorName(vs.ObjectMeta.Namespace, pl.Service, pl.Monitor.Type, pl.ServicePort, vs.Spec.Host, pl.Path)
 			}
 			pool.MonitorNames = append(pool.MonitorNames, MonitorName{Name: JoinBigipPath(rsCfg.Virtual.Partition, monitorName)})
 			monitor := Monitor{
@@ -463,7 +463,7 @@ func (ctlr *Controller) prepareRSConfigFromVirtualServer(
 						formatPort = pl.ServicePort
 					}
 					if monitor.Name == "" {
-						monitorName = formatMonitorName(vs.ObjectMeta.Namespace, pl.Service, monitor.Type, formatPort, monitor.Send)
+						monitorName = formatMonitorName(vs.ObjectMeta.Namespace, pl.Service, monitor.Type, formatPort, vs.Spec.Host, pl.Path)
 					}
 					pool.MonitorNames = append(pool.MonitorNames, MonitorName{Name: JoinBigipPath(rsCfg.Virtual.Partition, monitorName)})
 					monitor := Monitor{
@@ -1522,7 +1522,7 @@ func (ctlr *Controller) prepareRSConfigFromTransportServer(
 		pool.MonitorNames = append(pool.MonitorNames, MonitorName{Name: monitorName, Reference: vs.Spec.Pool.Monitor.Reference})
 	} else if vs.Spec.Pool.Monitor.Type != "" {
 		if vs.Spec.Pool.Name == "" {
-			monitorName = formatMonitorName(vs.ObjectMeta.Namespace, vs.Spec.Pool.Service, vs.Spec.Pool.Monitor.Type, vs.Spec.Pool.ServicePort, "")
+			monitorName = formatMonitorName(vs.ObjectMeta.Namespace, vs.Spec.Pool.Service, vs.Spec.Pool.Monitor.Type, vs.Spec.Pool.ServicePort, "", "")
 		}
 		pool.MonitorNames = append(pool.MonitorNames, MonitorName{Name: JoinBigipPath(rsCfg.Virtual.Partition, monitorName)})
 
@@ -1551,7 +1551,7 @@ func (ctlr *Controller) prepareRSConfigFromTransportServer(
 				}
 
 				if monitor.Name == "" {
-					monitorName = formatMonitorName(vs.ObjectMeta.Namespace, pl.Service, monitor.Type, formatPort, monitor.Send)
+					monitorName = formatMonitorName(vs.ObjectMeta.Namespace, pl.Service, monitor.Type, formatPort, "", "")
 				}
 				pool.MonitorNames = append(pool.MonitorNames, MonitorName{Name: JoinBigipPath(rsCfg.Virtual.Partition, monitorName)})
 				monitor := Monitor{
@@ -1652,9 +1652,9 @@ func (ctlr *Controller) prepareRSConfigFromLBService(
 			log.Errorf("[CORE] %s", msg)
 		}
 		pool.MonitorNames = append(pool.MonitorNames, MonitorName{Name: JoinBigipPath(rsCfg.Virtual.Partition,
-			formatMonitorName(svc.Namespace, svc.Name, monitorType, svcPort.TargetPort.IntVal, ""))})
+			formatMonitorName(svc.Namespace, svc.Name, monitorType, svcPort.TargetPort.IntVal, "", ""))})
 		monitor = Monitor{
-			Name:      formatMonitorName(svc.Namespace, svc.Name, monitorType, svcPort.TargetPort.IntVal, ""),
+			Name:      formatMonitorName(svc.Namespace, svc.Name, monitorType, svcPort.TargetPort.IntVal, "", ""),
 			Partition: rsCfg.Virtual.Partition,
 			Type:      monitorType,
 			Interval:  mon.Interval,

--- a/pkg/controller/resourceConfig_test.go
+++ b/pkg/controller/resourceConfig_test.go
@@ -105,8 +105,8 @@ var _ = Describe("Resource Config Tests", func() {
 			Expect(name).To(Equal("svc1_80_default_foo_app_test"), "Invalid Pool Name")
 		})
 		It("Monitor Name", func() {
-			name := formatMonitorName(namespace, "svc1", "http", 80, "")
-			Expect(name).To(Equal("svc1_default_http_80"), "Invalid Monitor Name")
+			name := formatMonitorName(namespace, "svc1", "http", 80, "foo.com", "path")
+			Expect(name).To(Equal("svc1_default_foo_com_path_http_80"), "Invalid Monitor Name")
 		})
 		It("Rule Name", func() {
 			name := formatVirtualServerRuleName("test.com", "", "", "sample_pool")

